src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+ 
+         // verify the previous addresses got Default cleared.
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c from Address__c];
+        // each set of edits created a new address object 
+        system.assertEquals(16, listAddr.size());
+        for (Address__c addr : listAddr) {
+            boolean fNewDefault = (addr.MailingStreet__c.contains('final Street Edit'));
+            system.assertEquals(fNewDefault, addr.Default_Address__c);
+        }                
+    }
+
+    /*********************************************************************************************************
+    @description
+        update Address' billing address and mark it as override, to cause creation of new non-default Address object  
+    verify:
+        new Address created
+        original Addresses still default    
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrMarkOverride() {
+        configSettings();
+             
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // modify the account addresses directly, and set override
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = 'Direct Street Edit';
+            accsAddrs.accs[i].BillingCity = 'Direct City Edit';
+            accsAddrs.accs[i].is_Address_Override__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update accsAddrs.accs;
+        Test.stopTest();
+        
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c from Account]);
+        
+        system.assertEquals('Direct Street Edit', mapAccIdAcc.values()[0].BillingStreet);
+        system.assertEquals('Direct City Edit', mapAccIdAcc.values()[0].BillingCity);
+        system.assertEquals(true, mapAccIdAcc.values()[0].is_Address_Override__c);
+        system.assertEquals('Direct Street Edit', mapAccIdAcc.values()[1].BillingStreet);
+        system.assertEquals('Direct City Edit', mapAccIdAcc.values()[1].BillingCity);
+        system.assertEquals(true, mapAccIdAcc.values()[1].is_Address_Override__c);
+ 
+         // verify the previous addresses didn't get Default cleared.
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.